--- a/mt7915/init.c
+++ b/mt7915/init.c
@@ -476,10 +476,9 @@
 				IEEE80211_VHT_CAP_MAX_MPDU_LENGTH_7991 |
 				IEEE80211_VHT_CAP_MAX_A_MPDU_LENGTH_EXPONENT_MASK;
 
-			if (!dev->dbdc_support)
-				vht_cap->cap |=
-					IEEE80211_VHT_CAP_SHORT_GI_160 |
-					FIELD_PREP(IEEE80211_VHT_CAP_EXT_NSS_BW_MASK, 1);
+			vht_cap->cap |=
+				IEEE80211_VHT_CAP_SHORT_GI_160 |
+				FIELD_PREP(IEEE80211_VHT_CAP_EXT_NSS_BW_MASK, 1);
 		} else {
 			phy->mt76->sband_5g.sband.ht_cap.ampdu_density =
 				IEEE80211_HT_MPDU_DENSITY_2;
@@ -494,8 +493,7 @@
 				IEEE80211_VHT_CAP_SUPP_CHAN_WIDTH_160MHZ;
 		}
 
-		if (!is_mt7915(&dev->mt76) || !dev->dbdc_support)
-			ieee80211_hw_set(hw, SUPPORTS_VHT_EXT_NSS_BW);
+		ieee80211_hw_set(hw, SUPPORTS_VHT_EXT_NSS_BW);
 	}
 
 	mt76_set_stream_caps(phy->mt76, true);
@@ -973,7 +971,7 @@
 		if (!dev->dbdc_support)
 			sts_160 /= 2;
 		else
-			sts_160 = 0;
+			sts_160 = 1;
 	}
 
 #ifdef CONFIG_MAC80211_MESH
@@ -1065,8 +1063,8 @@
 		/* Can do 1/2 of NSS streams in 160Mhz mode for mt7915 */
 		nss_160 = nss / 2;
 	else
-		/* Can't do 160MHz with mt7915 dbdc */
-		nss_160 = 0;
+		/* Allow 1x1 160MHz with mt7915 dbdc (1201 Mbps MCS 11) */
+		nss_160 = 1;
 
 	for (i = 0; i < 8; i++) {
 		if (i < nss)
